extensions/kafka/src/main/java/com/hazelcast/jet/kafka/impl/ResumeTransactionUtil.java

```diff
@@ -123,9 +116,75 @@ private static Object getValue(Object object, Class<?> clazz, String fieldName)
         }
     }
 
-    private static void setValue(Object object, String fieldName, Object value) {
+    private static Object createProducerIdAndEpoch(long producerId, short epoch) {
         try {
-            Field field = object.getClass().getDeclaredField(fieldName);
+            Field field =
+                    TransactionManager.class.getDeclaredField(PRODUCER_ID_AND_EPOCH_FIELD_NAME);
+            Class<?> clazz = field.getType();
+            Constructor<?> constructor = clazz.getDeclaredConstructor(Long.TYPE, Short.TYPE);
+            constructor.setAccessible(true);
```

Comment:
It works because Kafka dependency is not a JDK module so it's a part of the unnamed module (I guess fat/non-fat jar doesn't make any difference). As @viliam-durina told me, currently there's no clean way to resume Kafka transactions. I will run tests on JDK17 to be sure